From d82ab385c3411d5085f6c08871d6a7a118b7a419 Mon Sep 17 00:00:00 2001
From: effects3d <effects3dee@gmail.com>
Date: Sun, 13 Sep 2026 13:50:32 +0200
Subject: [PATCH 08/10] =?UTF-8?q?Revert=20=E2=80=9CHACK:=20win32u:=20Avoid?=
 =?UTF-8?q?=20loading=20EGL=20if=20WINE=5FUSE=5FEGL=3D0.=E2=80=9D?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Roblox Studio's performance under winex11's GLX backend is abysmal, so
use EGL by default.

Source: Proton-Wine commit: c3202e9d9f68191508892be146dab2e1ba929f78
Signed-off-by: effects3d <effects3dee@gmail.com>
---
 dlls/win32u/opengl.c | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/dlls/win32u/opengl.c b/dlls/win32u/opengl.c
index c7aeea130b8..e319289ad34 100644
--- a/dlls/win32u/opengl.c
+++ b/dlls/win32u/opengl.c
@@ -1433,13 +1433,7 @@ static const struct opengl_driver_funcs egldrv_funcs =
 static BOOL egl_init( const struct opengl_driver_funcs **driver_funcs )
 {
     struct opengl_funcs *funcs = &display_funcs;
-    const char *extensions, *env;
-
-    if (!(env = getenv( "WINE_USE_EGL" )) || !atoi( env ))
-    {
-        WARN( "EGL support is disabled.\n" );
-        return FALSE;
-    }
+    const char *extensions;
 
     if (!(funcs->egl_handle = dlopen( SONAME_LIBEGL, RTLD_NOW | RTLD_GLOBAL )))
     {
-- 
2.55.0

